python/ray/serve/scripts.py

```diff
@@ -504,39 +505,92 @@ def shutdown(address: str, yes: bool):
+        if not isinstance(node, (ClassNode, FunctionNode)):
+            raise TypeError(
+                f"Expected '{import_path}' to be ClassNode or "
+                f"FunctionNode, but got {type(node)}."
+            )
 
-    app = build_app(node)
-    schema = ServeApplicationSchema(
-        import_path=import_path,
-        runtime_env={},
-        host="0.0.0.0",
-        port=8000,
-        deployments=[deployment_to_schema(d) for d in app.deployments.values()],
-    )
+        app = build_app(node)
+        schema = ServeApplicationSchema(
+            import_path=import_path,
+            runtime_env={},
+            deployments=[
+                deployment_to_schema(d, not multi_app) for d in app.deployments.values()
+            ],
+        )
+        # If building a multi-app config, auto-generate names for each application.
+        # Also, each ServeApplicationSchema should not have host and port set, it should
+        # be set at the top level of ServeDeploySchema.
+        if multi_app:
+            schema.name = name
+            schema.route_prefix = app.ingress.route_prefix
+        else:
+            schema.host = "0.0.0.0"
+            schema.port = 8000
 
-    if kubernetes_format:
-        config = schema.kubernetes_dict(exclude_unset=True)
-    else:
-        config = schema.dict(exclude_unset=True)
+        if kubernetes_format:
+            return schema.kubernetes_dict(exclude_unset=True)
+        else:
+            return schema.dict(exclude_unset=True)
 
     config_str = (
         "# This file was generated using the `serve build` command "
         f"on Ray v{ray.__version__}.\n\n"
     )
-    config_str += yaml.dump(
-        config, Dumper=ServeBuildDumper, default_flow_style=False, sort_keys=False
-    )
+
+    if not multi_app:
+        if len(import_paths) > 1:
+            raise click.ClickException(
+                "Extra arguments detected. If you want to generate a multi-application "
+                "config, please turn on the feature flag `--multi-app`."
+            )
+
+        config_str += yaml.dump(
+            build_app_config(import_paths[0]),
+            Dumper=ServeApplicationSchemaDumper,
+            default_flow_style=False,
+            sort_keys=False,
+        )
+    else:
+        configs = []
+        for app_index, import_path in enumerate(import_paths):
+            configs.append(build_app_config(import_path, f"app{app_index + 1}"))
+
+        config_str += yaml.dump(
+            {
+                "host": "0.0.0.0",
+                "port": 8000,
+                "applications": configs,
+            },
+            Dumper=ServeDeploySchemaDumper,
+            default_flow_style=False,
+            sort_keys=False,
+        )
+        # NOTE(zcin): For some reason cli_logger.warning uses _level_str="WARN" instead
+        # of "WARNING", which doesn't stream this to stderr.
+        cli_logger._warning(
```

Comment:
Should this be an `INFO` log instead of a `WARNING`? It feels strange to raise a warning when everything in the command is working as expected.

Reply:
That's true, changed to info!